src/packaging/markers.py

```diff
@@ -243,10 +327,10 @@ def evaluate(self, environment: Optional[Dict[str, str]] = None) -> bool:
         current_environment = default_environment()
         current_environment["extra"] = ""
         if environment is not None:
-            current_environment.update(environment)
+            current_environment.update(environment)  # type: ignore[typeddict-item]
```

Comment:
What's causing this type failure?

Reply:
It's caused by the `environment` being a dict with arbitrary keys:

```
src/packaging/markers.py:330: error: Argument 1 to "update" of "TypedDict" has incompatible type "Dict[str, str]"; expected "TypedDict({'implementation_name'?: str, 'implementation_version'?: str, 'os_name'?: str, 'platform_machine'?: str, 'platform_release'?: str, 'platform_system'?: str, 'platform_version'?: str, 'python_full_version'?: str, 'platform_python_implementation'?: str, 'python_version'?: str, 'sys_platform'?: str, 'extra'?: Optional[str]})"  [typeddict-item]
```